internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -248,7 +255,92 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 			return fmt.Errorf("error creating address spec: %w", err)
 		}
 
-		// all done, terminate controller
-		return nil
+		return ctrl.pollConnection(ctx, logger)
+	}
+}
+
+func (ctrl *ManagerController) clearSiderolinkAddresses(ctx context.Context, r controller.Runtime) error {
+	list, err := safe.ReaderList[*network.AddressSpec](ctx, r, network.NewAddressSpec(network.ConfigNamespaceName, "").Metadata())
+	if err != nil {
+		return err
+	}
+
+	for iter := safe.IteratorFromList(list); iter.Next(); {
+		address := iter.Value()
+
+		if address.TypedSpec().LinkName == constants.SideroLinkName {
+			if destroyErr := r.Destroy(ctx, address.Metadata()); destroyErr != nil {
+				return destroyErr
+			}
+		}
+	}
+
+	return nil
+}
+
+// nodeUUID returns node UUID from the provided function or from SMBIOS.
+func (ctrl *ManagerController) nodeUUID() (string, error) {
+	if ctrl.NodeUUIDFunc != nil {
+		return ctrl.NodeUUIDFunc()
+	}
+
+	s, err := smbios.GetSMBIOSInfo()
```

Comment:
don't we have this info already popualted in the hardware info controller?

Reply:
switched to use it.